src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonMetadataServices.Helpers.cs

```diff
@@ -166,5 +166,63 @@ private static void PopulateProperties(JsonTypeInfo typeInfo, Func<JsonSerialize
 
             // NB we don't need to sort source gen properties here since they were already sorted at compile time.
         }
+
+        private static JsonPropertyInfo<T> CreatePropertyInfoCore<T>(JsonPropertyInfoValues<T> propertyInfoValues, JsonSerializerOptions options)
```

Comment:
Replaces a `JsonPropertyInfo<T>` constructor that was specifically used for populating sourcegen metadata.